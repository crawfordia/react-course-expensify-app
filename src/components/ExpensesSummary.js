import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => {
    const numExpenses = props.expenses.length;
    const total = numeral(getExpensesTotal(props.expenses) / 100).format('$0,0.00');
    return (
        <div>
        <p>
            {`Viewing ${numExpenses} expenses`}
            {numExpenses > 0 && ` totalling ${total}`}
        </p>
        </div>
    )
};

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);