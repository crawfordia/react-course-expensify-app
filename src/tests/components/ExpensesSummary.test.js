import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

let wrapper;

beforeEach(() => {
    wrapper = shallow(<ExpensesSummary expenses={[]}/>)
})

test('Should show count with no total for no expenses', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should show total for single expense', () => {
    wrapper.setProps({
        expenses: [expenses[0]]
    });

    expect(wrapper).toMatchSnapshot();
});

test('Should show total for multiple expenses', () => {
    wrapper.setProps({
        expenses
    });

    expect(wrapper).toMatchSnapshot();
});