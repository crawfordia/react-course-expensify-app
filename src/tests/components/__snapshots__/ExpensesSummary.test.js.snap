// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should show count with no total for no expenses 1`] = `
<div>
  <p>
    Viewing 0 expenses
  </p>
</div>
`;

exports[`Should show total for multiple expenses 1`] = `
<div>
  <p>
    Viewing 3 expenses
     totalling $1,141.95
  </p>
</div>
`;

exports[`Should show total for single expense 1`] = `
<div>
  <p>
    Viewing 1 expenses
     totalling $1.95
  </p>
</div>
`;
